### PR TITLE
remove typings, move to npm type definitions

### PR DESCRIPTION
**Pull request by FredKSchott**
Looks like we weren't really using our typings definitions much anyway :tada:

- remove typings
- update gulp/npm/travis tasks

/cc @rictic @justinfagnani

### CODE REVIEW

File: custom_typings/merge-stream.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+declare module "merge-stream" {
```
Comment (rictic) [inline custom_typings/merge-stream.d.ts]: Add a TODO to upstream please

File: custom_typings/merge-stream.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  module mergeStream {
+    interface MergedStream extends NodeJS.ReadWriteStream {
+      add: (source: NodeJS.ReadableStream) => MergedStream;
+    }
+	}
```
Comment (rictic) [inline custom_typings/merge-stream.d.ts]: > whitespace. this closing curly brace is over indented

> related: clang-format? automatically eliminate almost all alignment angst!

> see polymer-analyzer for how to set it up, it's very easy and very nice

File: src/bundle.ts [hunk 3]
```diff
@@ -15,12 +15,9 @@ import {Transform} from 'stream';
 import File = require('vinyl');
 import * as logging from 'plylog';
 import {urlFromPath} from './path-transformers';
-import {StreamAnalyzer, DepsIndex} from './analyzer';
-import {compose} from './streams';
+import {StreamAnalyzer} from './analyzer';
```
Comment (rictic) [inline src/bundle.ts]: Consider enabling the two new noUnused* typescript 2.0 compiler options to catch this stuff.

Comment (FredKSchott) [inline src/bundle.ts]: very cool. I just finished up another PR to add this to tslint, I'll add it to tsconfig in that PR as well

File: custom_typings/merge-stream.d.ts (hunk 1 above)
Comment (FredKSchott) [inline custom_typings/merge-stream.d.ts]: I actually just looked at this again and we don't actually need a `merge-stream` declaration (only used in gulpfile.js). Deleting this file but created https://github.com/DefinitelyTyped/DefinitelyTyped/pull/11452 to fix upstream

File: custom_typings/merge-stream.d.ts (hunk 2 above)
Comment (FredKSchott) [inline custom_typings/merge-stream.d.ts]: see above, deleting file!

**Comment by FredKSchott** [comment]
@rictic PTAL

File: custom_typings/merge-stream.d.ts (hunk 1 above)
Comment (rictic) [inline custom_typings/merge-stream.d.ts]: To get into `@types` you want to send a PR to the `types-2.0` branch, not master

Comment (FredKSchott) [inline custom_typings/merge-stream.d.ts]: ah, good to know. I'll rebase

**Pull request merged**